packages/grafana-ui/src/components/Card/Card.tsx

```diff
@@ -0,0 +1,227 @@
+  metadata?: ReactNode | ReactNode[];
+  /** Card description text */
+  description?: string;
+  /** List of tags to display in the card */
+  tags?: string[];
+  /** Optional callback for tag onclick event */
+  onTagClick?: OnTagClick;
+  /** Indicates if the card and all its actions can be interacted with */
+  disabled?: boolean;
+  /** Image or icon to be displayed on the let side of the card */
+  image?: ReactNode;
+  /** Main card actions **/
+  actions?: ReactElement[];
+  /** Right-side actions */
+  secondaryActions?: ReactElement[];
+  /** Link to redirect to on card click. If provided, the Card inner content will be rendered inside `a` */
+  href?: string;
+  /** On click handler for the Card */
+  onClick?: () => void;
+}
+
+export const Card: FC<Props> = ({
+  heading,
+  description,
+  metadata,
+  tags = [],
+  onTagClick,
+  disabled,
+  image,
+  actions = [],
+  tooltip,
+  secondaryActions = [],
+  tag,
+  href,
+  onClick,
+  className,
+  ...htmlProps
+}) => {
+  const hasActions = Boolean(actions.length || secondaryActions.length);
+  const disableHover = disabled || actions.length > 1;
+  const theme = useTheme();
+  const styles = getStyles(theme, disabled && !actions.length, disableHover || !onClick);
+  // Join meta data elements by '|'
+  const meta = useMemo(
+    () =>
+      Array.isArray(metadata)
+        ? (metadata as ReactNode[]).filter(Boolean).reduce((prev, curr, i) => [
+            prev,
+            <span key={`separator_${i}`} className={styles.separator}>
+              |
+            </span>,
+            curr,
+          ])
+        : metadata,
+    [metadata, styles.separator]
+  );
+  const onCardClick = useCallback(() => (disableHover ? () => {} : onClick), [disableHover, onClick]);
+
+  return (
+    <CardContainer
+      tooltip={tooltip}
+      tag={tag}
+      tabIndex={disableHover ? undefined : 0}
+      className={cx(styles.container, className)}
+      onClick={onCardClick}
+      {...htmlProps}
+    >
+      <CardInner href={href}>
+        {image && <div className={styles.media}>{image}</div>}
+        <div className={styles.inner}>
+          <div className={styles.heading}>{heading}</div>
+          {meta && <div className={styles.metadata}>{meta}</div>}
+          {!!tags.length && <TagList tags={tags} onClick={onTagClick} className={styles.tagList} />}
+          {description && <p className={styles.description}>{description}</p>}
+          {hasActions && (
+            <div className={styles.actionRow}>
+              {!!actions.length && (
+                <div className={styles.actions}>{actions.map(action => cloneElement(action, { disabled }))}</div>
+              )}
+              {!!secondaryActions.length && <div className={styles.secondaryActions}>{secondaryActions}</div>}
+            </div>
+          )}
+        </div>
+      </CardInner>
+      <div className={styles.overlay} />
```

Comment:
What is this `overlay`?

Reply:
It's used for disabling the whole card and also giving it a disabled background.
![Screenshot 2020-10-20 at 11 44 59](https://user-images.githubusercontent.com/8878045/96562673-b176a000-12c9-11eb-9930-5901d1775bec.png)

Reply:
Ah, ok. Do we need an additional dom element for that? Think you could just apply the appropriate styling to the root dom element of the card, couldn't you?

Reply:
Yeah, but we also need to disable all click events on the underlined elements, for ex. if the there's no or only one primary action. Could try to use pseudo elements for that but don't think it'll work.

Reply:
what about `pointer-events: none`?

Reply:
Good point! There's already `pointer-events: none` for the container in disabled state, so I guess just styling `::after` pseudo-selector instead of using an additional div will work here.